lib/Agrammon/DB/Datasets.pm6

```diff
@@ -5,33 +5,36 @@ use Agrammon::DB::User;
 use Agrammon::Timestamp;
 
 class Agrammon::DB::Datasets does Agrammon::DB {
-    has Str $.version;
+    has %.agrammon-variant;
     has Agrammon::DB::User    $.user;
     has Agrammon::DB::Dataset @.collection;
 
     method load {
         self.with-db: -> $db {
             my $username = $!user.username;
-            my $results = $db.query(q:to/DATASETS/, $username, $!version);
+            my $results = $db.query(q:to/DATASETS/, ~$username, %!agrammon-variant<version>, %!agrammon-variant<gui>, %!agrammon-variant<model>);
```

Comment:
...so that this could be done with a `|self!variant` also, for consistency

Reply:
Done.